presto-main/src/test/java/io/prestosql/server/TestDynamicFilterService.java

```diff
@@ -434,6 +435,75 @@ public void testReplicatedDynamicFilter()
     }
 
+    @Test
+    public void testDynamicFilterCancellation()
+    {
+        DynamicFilterService dynamicFilterService = new DynamicFilterService(new FeaturesConfig());
+        DynamicFilterId filterId = new DynamicFilterId("df");
+        Expression df1 = expression("DF_SYMBOL1");
+        QueryId queryId = new QueryId("query");
+        StageId stageId = new StageId(queryId, 0);
+        List<TaskId> taskIds = ImmutableList.of(new TaskId(stageId, 0), new TaskId(stageId, 1));
+
+        TestDynamicFiltersStageSupplier dynamicFiltersStageSupplier = new TestDynamicFiltersStageSupplier(RUNNING);
+        dynamicFiltersStageSupplier.addTasks(taskIds);
+        dynamicFilterService.registerQuery(queryId, dynamicFiltersStageSupplier, ImmutableSet.of(filterId), ImmutableSet.of(filterId), ImmutableSet.of());
+        ColumnHandle column = new TestingColumnHandle("probeColumnA");
+        DynamicFilter dynamicFilter = dynamicFilterService.createDynamicFilter(
+                queryId,
+                ImmutableList.of(new DynamicFilters.Descriptor(filterId, df1)),
+                ImmutableMap.of(
+                        Symbol.from(df1), column));
+        assertFalse(dynamicFilter.isBlocked().isDone());
+        assertFalse(dynamicFilter.isComplete());
+        assertEquals(dynamicFilter.getCurrentPredicate(), TupleDomain.all());
+
+        // assert initial dynamic filtering stats
+        DynamicFiltersStats stats = dynamicFilterService.getDynamicFilteringStats(queryId, session);
+        assertEquals(stats.getTotalDynamicFilters(), 1);
+        assertEquals(stats.getDynamicFiltersCompleted(), 0);
+        assertEquals(stats.getLazyDynamicFilters(), 1);
+        assertEquals(stats.getReplicatedDynamicFilters(), 0);
+
+        dynamicFiltersStageSupplier.storeSummary(
+                filterId,
+                new TaskId(stageId, 0),
+                singleValue(INTEGER, 1L));
+        dynamicFilterService.collectDynamicFilters();
+        assertEquals(dynamicFilter.getCurrentPredicate(), TupleDomain.all());
+
+        stats = dynamicFilterService.getDynamicFilteringStats(queryId, session);
+        assertEquals(stats.getDynamicFiltersCompleted(), 0);
+
+        // DynamicFilter future cancellation should not affect DynamicFilterService
+        CompletableFuture<?> isBlocked = dynamicFilter.isBlocked();
+        assertFalse(isBlocked.isDone());
+        assertFalse(isBlocked.cancel(false));
+        assertFalse(dynamicFilter.isBlocked().isDone());
+        assertFalse(dynamicFilter.isComplete());
+        stats = dynamicFilterService.getDynamicFilteringStats(queryId, session);
+        assertEquals(stats.getDynamicFiltersCompleted(), 0);
+
+        dynamicFiltersStageSupplier.storeSummary(
+                filterId,
+                new TaskId(stageId, 1),
+                singleValue(INTEGER, 2L));
+        dynamicFilterService.collectDynamicFilters();
+        assertTrue(dynamicFilter.isBlocked().isDone());
```

Comment:
use cancelled `isBlocked` instead of new future

Reply:
done